DataSource/Extension/Symfony/Form/Field/FormFieldExtension.php

```diff
@@ -41,26 +76,322 @@ public function getExtendedFieldTypes()
      */
     public function initOptions(FieldTypeInterface $field)
     {
-        if ($field->getComparison() == 'isNull') {
-            $field->getOptionsResolver()
-                ->setDefaults(array(
-                    'form_options' => array(
-                        'choices' => array(
-                            'null' => $this->translator->trans('datasource.form.choices.is_null', array(), 'DataSourceBundle'),
-                            'no_null' => $this->translator->trans('datasource.form.choices.is_not_null', array(), 'DataSourceBundle')
-                        )
-                    )
-                ));
-        } else if ($field->getType() == 'boolean') {
-            $field->getOptionsResolver()
-                ->setDefaults(array(
-                    'form_options' => array(
-                        'choices' => array(
-                            '1' => $this->translator->trans('datasource.form.choices.yes', array(), 'DataSourceBundle'),
-                            '0' => $this->translator->trans('datasource.form.choices.no', array(), 'DataSourceBundle')
-                        )
+        $field->getOptionsResolver()
+            ->setDefaults(array(
+                'form_filter' => true,
+                'form_options' => array(),
+                'form_from_options' => array(),
+                'form_to_options' =>array()
+            ))
+            ->setDefined(array(
+                'form_type',
+                'form_order'
+            ))
+            ->setAllowedTypes('form_filter', 'bool')
+            ->setAllowedTypes('form_options', 'array')
+            ->setAllowedTypes('form_from_options', 'array')
+            ->setAllowedTypes('form_to_options', 'array')
+            ->setAllowedTypes('form_order', 'integer')
+            ->setAllowedTypes('form_type', 'string')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function postBuildView(FieldEvent\ViewEventArgs $event)
+    {
+        $field = $event->getField();
+        $view = $event->getView();
+
+        if ($form = $this->getForm($field)) {
+            $view->setAttribute('form', $form->createView());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preBindParameter(FieldEvent\ParameterEventArgs $event)
+    {
+        $field = $event->getField();
+        $field_oid = spl_object_hash($field);
+        $parameter = $event->getParameter();
+
+        if (!$form = $this->getForm($field)) {
```

Comment:
I'd move this condition above the line `$field_oid = spl_object_hash($field);` to prevent unnecessary steps, also please remove assignment from the condition statement :)